src/common/hooks/use-sandbox-state.tsx

```diff
@@ -0,0 +1,128 @@
+import { useSelector } from 'react-redux';
+import { unwrapResult } from '@reduxjs/toolkit';
+
+import { useDispatch } from '@common/hooks/use-dispatch';
+import { RootState } from '@store';
+
+import {
+  selectAccountBalance,
+  selectAccountLoading,
+  selectAccountNonce,
+  selectAccountTransactions,
+  selectErrorState,
+  selectIdentity,
+  selectLastFetch,
+  selectLocalNonce,
+  setLocalNonce,
+  resetLocalNonce,
+  setIdentity,
+  setUserData,
+  generateIdentity,
+  fetchAccount,
+  requestFaucetFunds,
+  broadcastTransaction,
+  BroadcastTxOptions,
+  clearAccountError,
+} from '@store/sandbox';
+
+import { IdentityPayload, Account } from '@store/sandbox/types';
+import { UserData } from 'blockstack/lib/auth/authApp';
+import { UserSession } from 'blockstack/lib';
+import { identityStorage, USERNAME_COOKIE, IDENTITY_COOKIE, usernameStorage } from '@common/utils';
+
+interface SanboxStateValues {
+  lastFetch?: number;
+  loading: string;
+  balance?: string;
+  transactions?: any;
+  nonce?: number;
+  localNonce?: number;
+  identity?: IdentityPayload;
```

Comment:
So is this basically following the idea that the sandbox can use either a local keychain, or the authenticator? I'm not seeing that in the UI - maybe the UI just isn't there yet.

Reply:
Not quite sure what you mean, can you elaborate. Currently the sandbox only allows for keychains in app, and does not use the authenticator keychain.